src/apps/explorer/components/SummaryCardsWidget/VolumeChartWidget.tsx

```diff
@@ -0,0 +1,123 @@
+import { UTCTimestamp } from 'lightweight-charts'
+import React, { useRef, useState, useEffect } from 'react'
+import styled from 'styled-components'
+
+import { PeriodButton, VolumeChart, VolumeItem, VolumeDataResponse } from './VolumeChart'
+import volumeDataJson from './volumeData.json'
+
+const WrapperVolumeChart = styled.div`
+  height: 19.6rem;
+`
+export enum VolumePeriod {
+  DAILY = '1D',
+  WEEKLY = '1W',
+  MONTHLY = '1M',
+  YEARLY = '1Y',
+}
+
+type RawVolumeItem = Pick<VolumeItem, 'id'> & {
+  timestamp: string
+  volumeUsd: string
+}
+
+// TODO move builds to a file where The graph API is called
+export function buildVolumeData(
+  _data: RawVolumeItem[],
+  volumePeriod: VolumePeriod,
+): {
+  data: VolumeItem[]
+  currentVolume: number
+  changedVolume: number
+} {
+  const periods = {
+    [VolumePeriod.DAILY]: 7,
+    [VolumePeriod.WEEKLY]: 14,
+    [VolumePeriod.MONTHLY]: 30,
+    [VolumePeriod.YEARLY]: 365,
```

Comment:
I know we haven't yet integrated the sdk and built the queries, but these values are how many data points we have, right? If that's true and from that we discussed:
- Daily: 24 (1 per hour)
- Weekly: 168 (24x7)
- Montlhy: 30 (1 per day)
- Yearly: 365 (1 per day)

Reply:
> 1. Shouldn't the graph color be green here instead of red? 

I was taking the [last](https://github.com/cowprotocol/explorer/blob/37-volume-chart/src/apps/explorer/components/SummaryCardsWidget/VolumeChartWidget.tsx#L45)  and penultimate value of a [HistogramData](https://github.com/cowprotocol/explorer/blob/37-volume-chart/src/apps/explorer/components/SummaryCardsWidget/VolumeChart.tsx#L36) to make this comparison, I have changed it to the [first one](https://github.com/cowprotocol/explorer/blob/37-volume-chart/src/apps/explorer/components/SummaryCardsWidget/VolumeChartWidget.tsx#L46).

> 2. I believe we should limit how tall the graph can be so the labels don't collide with any line

@alongoni  what do you think if we reduce the [default size](https://github.com/cowprotocol/explorer/blob/37-volume-chart/src/apps/explorer/components/SummaryCardsWidget/VolumeChartWidget.tsx#L17) to 190 and align the chart downwards from the container?

> Daily: 24 (1 per hour)
Weekly: 168 (24x7)
Montlhy: 30 (1 per day)
Yearly: 365 (1 per day)

@ramirotw :point_up: Yes that would be what we discussed, however I have used ascending points just to emulate a change in graph. This will be finalized in the integration where a logic can be applied to request points **per day** or **per hour**